src/view/RepeatedFragment.js

```diff
@@ -210,6 +210,10 @@ export default class RepeatedFragment {
 			if ( newIndex === -1 ) return;
 
 			const fragment = this.iterations[ oldIndex ];
+
+			// guard against null fragment - #2005 (feels like treating the symptom, not the cause?)
+			if ( fragment == null ) return;
```

Comment:
I think I'm too much of a noob to tackle the delegation, which sounds like the right approach. For example, how would one build the keypath to invoke the native methods? From the template? Sorry, you have alot on your plate already, but it's probably better if you do a PR. I'll help where I can.